File: Longest K unique characters substring.java
class Solution {
    public int longestkSubstr(String s, int k) {
        // code here
        int j =0, i =0; 
        HashMap<Character, Integer> map = new HashMap<>();
        int max = -1;
        while (j<s.length()){
            map.put(s.charAt(j),map.getOrDefault(s.charAt(j), 0)+1 );
            
            
            while (map.size()>k){
            map.put(s.charAt(i),map.getOrDefault(s.charAt(i), 0)-1 );
                if (map.get(s.charAt(i))==0){
                    map.remove(s.charAt(i));
                }
                i++;
            }
            if (map.size()==k){
                max = Math.max(max, j-i+1);
            }
            j++;
        }
        return max;
    }
}

File: Longest K unique characters substring.cpp
  int longestKSubstr(string s, int k) {
    // your code here
    int i=0;
    int res=-1;
    unordered_map<char,int>mp;
    for(int j=0;j<s.length();j++){
        mp[s[j]]++;
        while(mp.size()>k){
            mp[s[i]]--;
            if(mp[s[i]]==0){
                mp.erase(s[i]);
            }
            i++;
        }
        if(mp.size()==k){
            res=max(res,j-i+1);
        }
    }
    return res;
    
